model/k8s/metrics.yaml

```
@@ -469,3 +469,193 @@ groups:
+    type: metric
+    metric_name: k8s.resourcequota.memory.limit
+    stability: development
+    brief: |
+      The memory limits in a specific namespace.
+      See [`k8s.resourcequoate.state`](../registry/attributes/k8s.md) attribute for further details.
+    instrument: gauge
+    unit: "By"
+    note: |
+      More info: https://kubernetes.io/docs/concepts/policy/resource-quotas/#compute-resource-quota
+
+      This metric SHOULD, at a minimum, be reported against a
+      [`k8s.resourcequota`](../resource/k8s.md#resourcequota) resource.
+    attributes:
+      - ref: k8s.resourcequota.state
+        requirement_level: required
+  - id: metric.k8s.resourcequota.memory.request
+    type: metric
+    metric_name: k8s.resourcequota.memory.request
+    stability: development
+    brief: |
+      The memory requests in a specific namespace.
+      See [`k8s.resourcequoate.state`](../registry/attributes/k8s.md) attribute for further details.
+    instrument: gauge
+    unit: "By"
+    note: |
+      More info: https://kubernetes.io/docs/concepts/policy/resource-quotas/#compute-resource-quota
+
+      This metric SHOULD, at a minimum, be reported against a
+      [`k8s.resourcequota`](../resource/k8s.md#resourcequota) resource.
+    attributes:
+      - ref: k8s.resourcequota.state
+        requirement_level: required
+
+  # k8s.resourcequota.hugepages.* metrics
+  - id: metric.k8s.resourcequota.hugepages.request
+    type: metric
+    metric_name: k8s.resourcequota.hugepages.request
+    stability: development
+    brief: |
+      The huge page requests in a specific namespace.
+      See [`k8s.resourcequoate.state`](../registry/attributes/k8s.md) attribute for further details.
+    instrument: gauge
+    unit: "By"
+    note: |
+      More info: https://kubernetes.io/docs/concepts/policy/resource-quotas/#compute-resource-quota
+
+      This metric SHOULD, at a minimum, be reported against a
+      [`k8s.resourcequota`](../resource/k8s.md#resourcequota) resource.
+    attributes:
+      - ref: k8s.resourcequota.state
+        requirement_level: required
+      - ref: k8s.hugepage.size
+        requirement_level: required
+
+  # k8s.resourcequota.storage.* metrics
+  - id: metric.k8s.resourcequota.storage.request
+    type: metric
+    metric_name: k8s.resourcequota.storage.request
+    stability: development
+    brief: |
+      The storage requests in a specific namespace.
+      See [`k8s.resourcequoate.state`](../registry/attributes/k8s.md) attribute for further details.
+    instrument: gauge
+    unit: "By"
+    note: |
+      More info: https://kubernetes.io/docs/concepts/policy/resource-quotas/#storage-resource-quota
+
+      This metric SHOULD, at a minimum, be reported against a
+      [`k8s.resourcequota`](../resource/k8s.md#resourcequota) resource.
+    attributes:
+      - ref: k8s.resourcequota.state
+        requirement_level: required
+      - ref: k8s.storage.class_name
+        requirement_level: opt_in
```

```suggestion
        requirement_level: conditionally_required
```

When quotas are set both for generic storage (e.g., requests.storage) and storage class specific resources (e.g., example.storageclass.storage.k8s.io/requests.storage) in the same namespace, omitting the storage class name results in two indistinguishable metrics. If this metric is enabled, then the class_name should be required when a resourcequota is defined for storageclass.